db/model/archival/archivalDb.ts

```diff
@@ -566,6 +566,35 @@ export const getImagesByPostId = async (
     return imagesByPostId
 }
 
+export const getVideosByPostId = async (
+    knex: db.KnexReadonlyTransaction,
+    postIds: string[]
+): Promise<Record<string, string[]>> => {
+    if (postIds.length === 0) return {}
+
+    // We only care about video URLs for now and don't support re-archiving when
+    // a different video is uploaded with the same URL.
+    const videoComponents = await db.knexRaw<{
+        gdocId: string
+        url: string
+    }>(
+        knex,
+        `-- sql
+        SELECT
+            gdocId,
+            config ->> '$.url' AS url
+        FROM posts_gdocs_components
+        WHERE gdocId IN (${postIds.map(() => "?").join(",")})
+        AND config ->> '$.type' = 'video'
```

Comment:
Nothing you need to do about that now, but just appears to me:
We should turn `$.type` into a computed column in mysql, with an index on it for fast access.

Reply:
now up at #5575.